bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BIntersectionType.java

```diff
@@ -81,6 +84,20 @@ public int getTag() {
 
     @Override
     public String toString() {
+        if (resolving) {
```

Comment:
Can you share an example of when this fails for the intersection type rather than the effective type?

Reply:
I have to rework the code experiencing some issues in debugging. Almost like race conditions.
Check the same example in the original problem. The return type of the method and parameter is the intersection type of the object.
So this stack overflow loops over Object and Intersection both. Intersection type having a collection of other types is easily susceptible to this issue. That is why I decided to fix it there too.

<img width="586" alt="Screenshot 2021-06-21 at 1 27 40 PM" src="https://user-images.githubusercontent.com/2173530/122727276-80090680-d294-11eb-86cc-c1443bc1f9c3.png">

Reply:
But shouldn't this get fixed automatically when the object case is fixed?

Reply:
Missed the comment.
What if this also happens with records?

Reply:
Missed the comment.
What if this also happens with records?

Reply:
Missed the comment.
What if this also happens with records?

Reply:
Missed the comment.
What if this also happens with records?

Reply:
I will check if this is redundant and revert the change in another PR.